src/go/rpk/pkg/cli/cmd/container/start.go

```diff
@@ -316,6 +316,9 @@ func writeNodeConfig(
 	conf := config.DefaultConfig()
 	conf.Redpanda.Id = int(nodeID)
 
+	conf.Redpanda.AdvertisedKafkaApi = &config.SocketAddress{}
+	conf.Redpanda.AdvertisedRPCAPI = &config.SocketAddress{}
```

Comment:
isn't this the main problem, that they are not null

Reply:
Nope. `rpk container` never left them as null. The problem appeared when running it standalone.